dev/com.ibm.websphere.appserver.features/test/src/com/ibm/ws/feature/tests/VisibilityTest.java

```diff
@@ -283,4 +288,69 @@ public void testVisibility2() {
+    public void testAutoFeatures() {
+        StringBuilder errorMessage = new StringBuilder();
+        for (Entry<String, FeatureInfo> entry : features.entrySet()) {
+            String featureName = entry.getKey();
+            FeatureInfo featureInfo = entry.getValue();
+
+            if (!featureInfo.isAutoFeature()) {
+                continue;
+            }
+
+            String[] filterFeatures = featureInfo.getAutoFeatures();
+            if (filterFeatures.length == 1) {
+                errorMessage.append("Found issues with " + featureName + '\n');
+                errorMessage.append("     Auto feature filter only depends on one feature " + filterFeatures[0] + ".\n");
+                errorMessage.append("     The feature and/or bundle dependencies in this auto feature should just be a dependency of that feature\n");
+                errorMessage.append("     OR this should be turned into a private feature that " + filterFeatures[0] + " depends on.");
+            }
+        }
+        if (errorMessage.length() != 0) {
+            Assert.fail("Found auto features that have edition set incorrectly: " + '\n' + errorMessage.toString());
+        }
+    }
+
+    /**
+     * This test makes sure that public features have properties files that match the long
+     * feature name.  When moving features to the io.openliberty prefix from com.ibm.websphere.appserver
+     * and vice versa, the properties file renames were missed a few times.  This unit test
+     * makes sure that it is found in the build instead of having to be detected by hand.
+     */
+    @Test
+    public void testLocalizationResources() {
+        StringBuilder errorMessage = new StringBuilder();
+        for (Entry<String, FeatureInfo> entry : features.entrySet()) {
+            String featureName = entry.getKey();
+            FeatureInfo featureInfo = entry.getValue();
+
+            if (!"public".equals(featureInfo.getVisibility())) {
+                continue;
+            }
+
+            String shortName = featureInfo.getShortName();
+            File featureFile = featureInfo.getFeatureFile();
+
+            String fileName = featureFile.getAbsolutePath();
+            fileName = fileName.replace('\\', '/');
+
+            int lastSlash = fileName.lastIndexOf('/'); 
+            String expectedFileName = fileName.substring(0, lastSlash + 1) + "/resources/l10n/" + featureName + ".properties";
+            
+            if (!new File(expectedFileName).exists()) {
+                String expectedPropertiesFileRelPath = shortName + "/resources/l10n/" + featureName + ".properties"; 
+                errorMessage.append("Found issues with " + featureName + '\n');
+                errorMessage.append("     Expected to find file " + expectedPropertiesFileRelPath + '\n');
+            }
+        }
+        
+        if (errorMessage.length() != 0) {
+            Assert.fail("Found auto features that have edition set incorrectly: " + '\n' + errorMessage.toString());
```

Comment:
Again, seems to be a copy/paste issue.... perhaps "Found public features that properties files named incorrectly`?